### ISSUE TITLE
Site:GroundDomain outputs not in RDD

### ISSUE THREAD

**Issue by FarhangTahmasebi** [NREL/EnergyPlus#5321]
Modeling a grade slab using Site:GroundDomain:Slab, I realized that the Site:GroundDomain outputs (such as Zone Coupled Surface Temperature) are not placed in RDD file, even though they can be obtained from the model run. Using Output:Diagnostics with Display Advanced Report Variable does not change this.

**Comment by mjwitte** [comment]
This should not be possible, but I can reproduce it with any of the example files that use Site:GroundDomain:\* (ZoneCoupledGroundHTBasement, ZoneCoupledGroundHTSlabInGrade, 5ZoneAirCooledWithCoupledInGradeSlab).  There are documentation issues as well, and these output variables should be renamed to show that they go with the grounddomain model.
- [x] Fix missing names in rdd
- [x] Consider renaming these output variable to begin with Ground Domain . . .
- [x] Input Output Reference section titled "Site:GroundDomain Outputs" shows only the slab outputs, the basement ones are not documented.  Either move this section after Site:GroundDomain:Basement (and add the basement outputs) or rename this section to Site:GroundDomain:Slab Outputs and make a new section for the basement outputs.
- [x] The descriptions of the slab outputs have typos "OtherSideBoudaryCondition model" and "OthersideConditionModel" suggest changing these to "SurfaceProperty:OtherSideConditionsModel", and in the first one "domains" should be "domain's" or rephrase the sentence.
- [x] Example file ZoneCoupledGroundHTBasement currently has the variable dictionary commented out and doesn't even use the grounddomain:basement outputs.  These should be added and the dictionary uncommented.

**Comment by mjwitte** [comment]
@mmatts FYI.

**Comment by mitchute** [comment]
@Myoldmopar @mjwitte The RDD entries are missing because [SetupOutputVariables](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/PlantPipingSystemsManager.cc#L2339) for the slab and basement outputs is being called after [ProduceRDDMDD()](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/SimulationManager.cc#L400). Would it be possible to move this to after the first timestep has occurred? I am guessing that this problem is not limited to this model, unless there is another way I missed.

Adding

``` c++
static bool writeRDDMDD( true );
```

and moving `ProduceRDDMDD()` to the end of the timestep loop as below flushes them to the RDD.

``` c++
if ( writeRDDMDD ) {
     ProduceRDDMDD();
     writeRDDMDD = false;
}
```

Any suggestions?

**Comment by mjwitte** [comment]
@mmatts The comment a few lines above ProduceRDDMDD() says " Note:  All the inputs have been 'gotten' by the time we get here"  Have the inputs for Site:GroundDomain been read by this point?  If not, they should be.  If they've been gotten, then the SetupOutputVariable calls need to be moved to happen as soon as the ground domain input is read.  Most modules set these up right in the getinput routine.

**Comment by mitchute** [comment]
@mjwitte Thanks for pointing that out. [Slab and basement inputs](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/SimulationManager.cc#L507) were not being called until later. I added the getInput to [SetupSimulation](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/SimulationManager.cc#L1511) and that cures the problem.

**Comment by Myoldmopar** [comment]
Fixed via #5323
